src/board.rs

```diff
@@ -52,8 +52,12 @@ impl Board {
         ]
     }
 
-    pub fn get_piece(&self, pos: &Position) -> Option<Piece> {
-        self.squares[pos.y][pos.x]
+    pub fn get_piece(&self, pos: &Position) -> &Option<Piece> {
```

Comment:
What's the argument for returning `&Option` here?

Reply:
probably just borrow until we need to clone

Reply:
Isn't `Option<Piece>` smaller than a reference?

Reply:
so it does matter?

Reply:
It either does it which case we should copy to save memory, or it doesn't in which case we should copy to make the code easier to work with.